File: Client - Java/src/main/java/controller/ISellTicketsServer.java
package controller;


import entity.Match;
import entity.User;
import exceptions.ControllerException;

import java.rmi.Remote;
import java.rmi.RemoteException;
import java.util.List;

public interface ISellTicketsServer extends Remote {
    public void login(User user, ISellTicketsClient client) throws SaleHouseException;
    public void logout(User user, ISellTicketsClient client) throws SaleHouseException;
    public void sellTickets(String idMatch, String quantity, String buyerPerson, String username) throws SaleHouseException;

    List<Match> getAllMatches() throws SaleHouseException;
    List<Match> getFilteredAndSortedMatches() throws SaleHouseException;
    void setClient(ISellTicketsClient client);
}

File: Server - CS/Server/service/ISellTicketsServer.cs
using System;
using System.Collections.Generic;
using Server.domain;

namespace Server.service
{
    public interface ISellTicketsServer
    {
        void Login(User user, ISellTicketsClient client);
        void Logout(User user, ISellTicketsClient client);
        void SellTickets(String idMatch, String quantity, String buyerPerson, String username);

        List<Match> GetAllMatches() ;
        List<Match> GetFilteredAndSortedMatches();

    }
}
